digdag-standards/src/main/java/io/digdag/standards/operator/aws/AWSSessionCredentialsFactory.java

```diff
@@ -0,0 +1,176 @@
+package io.digdag.standards.operator.aws;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.auth.policy.Policy;
+import com.amazonaws.auth.policy.Resource;
+import com.amazonaws.auth.policy.Statement;
+import com.amazonaws.auth.policy.actions.DynamoDBv2Actions;
+import com.amazonaws.auth.policy.actions.ElasticMapReduceActions;
+import com.amazonaws.auth.policy.actions.S3Actions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleResult;
+import com.amazonaws.services.securitytoken.model.Credentials;
+import com.amazonaws.services.securitytoken.model.GetFederationTokenRequest;
+import com.amazonaws.services.securitytoken.model.GetFederationTokenResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class AWSSessionCredentialsFactory
+{
+    private static final int DEFAULT_DURATION_SECONDS = 3 * 3600;
+    private static final String DEFAULT_SESSION_NAME = "digdag-operator-session";
+    private static final String URI_S3_PREFIX = "s3://";
+    private static final String URI_DYNAMODB_PREFIX = "dynamodb://";
+    private static final String URI_EMR_PREFIX = "emr://";
+
+    private final String accessKeyId;
+    private final String secretAccessKey;
+    private final List<AcceptableUri> acceptableUris;
+    private String roleArn;
+    private String sessionName = DEFAULT_SESSION_NAME;
+    private int durationSeconds = DEFAULT_DURATION_SECONDS;
+
+    public enum Mode
+    {
+        READ, WRITE
+    }
+
+    public static class AcceptableUri
+    {
+        private final Mode mode;
+        private final String uri;
+
+        public AcceptableUri(Mode mode, String uri)
+        {
+            this.mode = mode;
+            this.uri = uri;
+        }
+    }
+
+    public AWSSessionCredentialsFactory(String accessKeyId, String secretAccessKey, List<AcceptableUri> acceptableUris)
```

Comment:
I think you can use com.amazonaws.auth.BasicAWSCredentials instead of accessKeyId + secretAccessKey. Then, BaseRedshiftLoadOperator.createBaseCredential returns BasicAWSCredentials and code will be simpler.